[Bug]: terraform-provider-aws_v5.40.0_x5 plugin crash. caused by empty aws_ecs_cluster configuration block

### Terraform Core Version

1.5.7,1.7.4

### AWS Provider Version

v5.40.0_x5

### Affected Resource(s)

aws_v5.40.0_x5

### Expected Behavior

Terraform plan passed successfully and terraform apply validation step passed, terraform apply create step should have completed.

### Actual Behavior

Error: The terraform-provider-aws_v5.40.0_x5 plugin crashed!


### Relevant Error/Panic Output Snippet

```shell
Stack trace from the terraform-provider-aws_v5.40.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 47 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ecs.expandClusterConfigurationExecuteCommandLogConfiguration({0x14003f88690?, 0x1400273e8a0?, 0x111442af4?})
	github.com/hashicorp/terraform-provider-aws/internal/service/ecs/cluster.go:618 +0x368
github.com/hashicorp/terraform-provider-aws/internal/service/ecs.expandClusterConfigurationExecuteCommandConfiguration({0x14003f88670?, 0x1400273e000?, 0x111500d91?})
	github.com/hashicorp/terraform-provider-aws/internal/service/ecs/cluster.go:600 +0xb4
github.com/hashicorp/terraform-provider-aws/internal/service/ecs.expandClusterConfiguration({0x14003f88640?, 0x1114117a3?, 0x140026c9b30?})
	github.com/hashicorp/terraform-provider-aws/internal/service/ecs/cluster.go:586 +0xb4
github.com/hashicorp/terraform-provider-aws/internal/service/ecs.resourceClusterCreate({0x1161f5788, 0x140026c9b30}, 0x1400380ea80, {0x115fd17c0?, 0x14002ef5180})
	github.com/hashicorp/terraform-provider-aws/internal/service/ecs/cluster.go:171 +0x194
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0x0?, {0x115fd17c0?, 0x14002ef5180?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:112 +0x1d4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1161f5788?, {0x1161f5788?, 0x14002681560?}, 0xd?, {0x115fd17c0?, 0x14002ef5180?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:773 +0x64
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000d50d20, {0x1161f5788, 0x14002681560}, 0x14003cda680, 0x1400380e900, {0x115fd17c0, 0x14002ef5180})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:909 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14002f06138, {0x1161f5788?, 0x140026813e0?}, 0x14003d5ce10)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xb08
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x1161f57c0?, {0x1161f5788?, 0x14002680a20?}, 0x14003d5ce10)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140008f2000, {0x1161f5788?, 0x14002680120?}, 0x140008f0850)
	github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/tf5server/server.go:846 +0x2b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x115d7d1a0?, 0x140008f2000}, {0x1161f5788, 0x14002680120}, 0x1400380e500, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140016b0000, {0x1161f5788, 0x14002680000}, {0x11622b3f8, 0x140030c29c0}, 0x1400366ea20, 0x140013642d0, 0x11e079a58, 0x0)
	google.golang.org/grpc@v1.62.0/server.go:1383 +0xb8c
google.golang.org/grpc.(*Server).handleStream(0x140016b0000, {0x11622b3f8, 0x140030c29c0}, 0x1400366ea20)
	google.golang.org/grpc@v1.62.0/server.go:1794 +0xc70
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.62.0/server.go:1027 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 41
	google.golang.org/grpc@v1.62.0/server.go:1038 +0x150

Error: The terraform-provider-aws_v5.40.0_x5 plugin crashed!
```


### Terraform Configuration Files

This configuration triggered the crash:
resource "aws_ecs_cluster" "cluster_prod" {
  name = "my-cluster-name"
  configuration {
    execute_command_configuration {
      log_configuration {
      }
    }
  }
  setting {
    name  = "containerInsights"
    value = "enabled"
  }
}
Note that the log_configuration is empty.

### Steps to Reproduce

Running this configuration passes validation but crashes when applied. Removing this block allows the cluster to be created:
  configuration {
    execute_command_configuration {
      log_configuration {
      }
    }


### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/36341.txt
+++ b/.changelog/36341.txt
@@ -1,3 +1,7 @@
 ```release-note:bug
-resource/aws_ecs_cluster: Add nil checks for the `configuration`, `execute_command_configuration`, and `log_configuration` blocks to prevent provider crashes when empty blocks are supplied
+resource/aws_ecs_cluster: Fix `panic: interface conversion: interface {} is nil, not map[string]interface {}` when `configuration`, `configuration.execute_command_configuration`, or `configuration.execute_command_configuration.log_configuration` are empty
+```
+
+```release-note:enhancement
+resource/aws_ecs_cluster: Add default value (`DEFAULT`) for `configuration.execute_command_configuration.logging`
 ```

--- a/.changelog/36341.txt
+++ b/.changelog/36341.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_ecs_cluster: Add nil checks for the `configuration`, `execute_command_configuration`, and `log_configuration` blocks to prevent provider crashes when empty blocks are supplied
+```

--- a/internal/service/ecs/cluster.go
+++ b/internal/service/ecs/cluster.go
@@ -93,6 +93,7 @@ func ResourceCluster() *schema.Resource {
 									"logging": {
 										Type:         schema.TypeString,
 										Optional:     true,
+										Default:      ecs.ExecuteCommandLoggingDefault,
 										ValidateFunc: validation.StringInSlice(ecs.ExecuteCommandLogging_Values(), false),
 									},
 								},
@@ -576,7 +577,7 @@ func flattenClusterConfigurationExecuteCommandConfigurationLogConfiguration(apiO
 }
 
 func expandClusterConfiguration(nc []interface{}) *ecs.ClusterConfiguration {
-	if len(nc) == 0 {
+	if len(nc) == 0 || nc[0] == nil {
 		return &ecs.ClusterConfiguration{}
 	}
 	raw := nc[0].(map[string]interface{})
@@ -590,7 +591,7 @@ func expandClusterConfiguration(nc []interface{}) *ecs.ClusterConfiguration {
 }
 
 func expandClusterConfigurationExecuteCommandConfiguration(nc []interface{}) *ecs.ExecuteCommandConfiguration {
-	if len(nc) == 0 {
+	if len(nc) == 0 || nc[0] == nil {
 		return &ecs.ExecuteCommandConfiguration{}
 	}
 	raw := nc[0].(map[string]interface{})
@@ -612,7 +613,7 @@ func expandClusterConfigurationExecuteCommandConfiguration(nc []interface{}) *ec
 }
 
 func expandClusterConfigurationExecuteCommandLogConfiguration(nc []interface{}) *ecs.ExecuteCommandLogConfiguration {
-	if len(nc) == 0 {
+	if len(nc) == 0 || nc[0] == nil {
 		return &ecs.ExecuteCommandLogConfiguration{}
 	}
 	raw := nc[0].(map[string]interface{})

